src/Microsoft.Azure.ServiceBus/SessionReceivePump.cs

```diff
@@ -93,10 +72,10 @@ bool ShouldRenewSessionLock()
                 this.sessionHandlerOptions.AutoRenewLock;
         }
 
-        void RaiseExceptionReceived(Exception e, string action)
+        async Task RaiseExceptionReceived(Exception e, string action)
         {
-            var eventArgs = new ExceptionReceivedEventArgs(e, action, this.endpoint, this.entityPath);
-            this.sessionHandlerOptions.RaiseExceptionReceived(eventArgs);
+            var eventArgs = new ExceptionReceivedEventArgs(e, action, this.endpoint, this.entityPath, this.clientId);
+            await this.sessionHandlerOptions.RaiseExceptionReceived(eventArgs).ConfigureAwait(false);
```

Comment:
It it's invoked and no need in the awaited result, method can drop `async` and return the task

`return this.sessionHandlerOptions.RaiseExceptionReceived(eventArgs)`